rclcpp/test/rclcpp/test_type_support.cpp

```diff
@@ -0,0 +1,201 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <memory>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/type_support_decl.hpp"
+#include "test_msgs/msg/empty.hpp"
+
+class TestTypeSupport : public ::testing::Test
+{
+public:
+  static void SetUpTestCase()
+  {
+    if (!rclcpp::ok()) {
+      rclcpp::init(0, nullptr);
+    }
+  }
+
+  static void TearDownTestCase()
+  {
+    rclcpp::shutdown();
+  }
+
+  ::testing::AssertionResult test_type_support_init_fini(const rosidl_service_type_support_t * ts)
+  {
+    rcl_service_t service_handle = rcl_get_zero_initialized_service();
+    rcl_service_options_t service_options = rcl_service_get_default_options();
+    rcl_ret_t ret = rcl_service_init(
+      &service_handle,
+      node->get_node_base_interface()->get_rcl_node_handle(),
+      ts, "base_node_service", &service_options);
+    if (ret != RCL_RET_OK) {
+      return ::testing::AssertionFailure() << "Failed ts init";
```

Comment:
I think this error message could be a little more descriptive by including an indication it was `rcl_service_init` that failed and what current rcl error string is. Same in line 54.

Reply:
See 2c9b77e